# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/DOMIntersectionObserver.cpp
# Commit: 9cc914a73a3f
# Full Hash: 9cc914a73a3f15e14df950a52e18da23791ca088
# Author: Daniel Holbert <dholbert@cs.stanford.edu>
# Date: 2021-03-20 08:56:43
# Regressor Bug: 1599795
# File Overlap Count: 1
# Description:
#   Bug 1699853: Use "InProcess" version of GetCrossDocParentFrame() in DOMIntersectionObserver.cpp. r=emilio
#   
#   This patch doesn't change behavior; it's just switching us between two
#   functions that do the same thing. (One is literally a trivial wrapper for the
#   other.)
# ==============================================================================

diff -r c42d8d55fea5 -r 9cc914a73a3f dom/base/DOMIntersectionObserver.cpp
--- a/dom/base/DOMIntersectionObserver.cpp	Fri Mar 19 23:26:21 2021 +0000
+++ b/dom/base/DOMIntersectionObserver.cpp	Fri Mar 19 23:58:30 2021 +0000
@@ -354,7 +354,8 @@
   // apply clip-path.
   //
   // 3. While container is not the intersection root:
-  nsIFrame* containerFrame = nsLayoutUtils::GetCrossDocParentFrame(target);
+  nsIFrame* containerFrame =
+      nsLayoutUtils::GetCrossDocParentFrameInProcess(target);
   while (containerFrame && containerFrame != aRoot) {
     // FIXME(emilio): What about other scroll frames that inherit from
     // nsHTMLScrollFrame but have a different type, like nsListControlFrame?
@@ -390,7 +391,8 @@
       target = containerFrame;
     }
 
-    containerFrame = nsLayoutUtils::GetCrossDocParentFrame(containerFrame);
+    containerFrame =
+        nsLayoutUtils::GetCrossDocParentFrameInProcess(containerFrame);
   }
   MOZ_ASSERT(intersectionRect);
 
